stacks/spica/packages/material/clipboard/clipboard.directive.ts

```diff
@@ -8,19 +8,31 @@ export class MatClipboardDirective {
   @Input() icon: string = "info";
   @Input("matClipboard") text: string;
 
+  clipBoardElement: HTMLInputElement;
+
   @HostListener("click")
   copy() {
     if (!this.text) {
       return;
     }
-    const clipboard = document.createElement("input");
-    clipboard.innerText = this.text;
-    clipboard.select();
-    clipboard.setSelectionRange(0, this.text.length);
+
+    this.prepareClipBoard(this.text);
+
     document.execCommand("copy");
+
+    document.body.removeChild(this.clipBoardElement);
+
     this.icon = "check";
     setTimeout(() => {
       this.icon = "info";
     }, 1000);
   }
+
+  prepareClipBoard(text: string) {
```

Comment:
From the functional programming perspective, this method can return the newly created element.

Reply:
Also, if we return the created element, we can eliminate `clipBoardElement` property. Much cleaner IMHO.